### ISSUE TITLE
New Chameleon update breaks websites

### ISSUE THREAD

**Issue by Minimalist73** [sereneblue/chameleon#476]
## Expected Behavior
Load the page correctly
## Current Behavior
Load the page, then breaks things or go completely blank. Does that on Twitch (on a livestream / logged-in), some sub-reddit (go blank) and can't start a search on google.com (need to hit enter multiple times).
## Relevant settings

Settings:
```
{
  "config": {
    "enabled": true,
    "notificationsEnabled": true,
    "theme": "dark",
    "hasPrivacyPermission": true
  },
  "excluded": [
    "win1-gcr",
    "win2-gcr",
    "win3-gcr",
    "win4-gcr",
    "win4-edg",
    "win2-ie",
    "win1-ie",
    "win3-ie",
    "win1-esr",
    "win3-esr",
    "win4-esr",
    "win4-ie"
  ],
  "headers": {
    "blockEtag": false,
    "enableDNT": false,
    "referer": {
      "disabled": false,
      "xorigin": "0",
      "trimming": "0"
    },
    "spoofAcceptLang": {
      "enabled": false,
      "value": "en-US"
    },
    "spoofIP": {
      "enabled": false,
      "option": 0,
      "rangeFrom": "",
      "rangeTo": ""
    }
  },
  "ipRules": [],
  "options": {
    "cookieNotPersistent": true,
    "cookiePolicy": "reject_third_party",
    "blockMediaDevices": false,
    "blockCSSExfil": false,
    "disableWebRTC": true,
    "firstPartyIsolate": true,
    "limitHistory": false,
    "protectKBFingerprint": {
      "enabled": false,
      "delay": 30
    },
    "protectWinName": false,
    "resistFingerprinting": false,
    "screenSize": "default",
    "spoofAudioContext": false,
    "spoofClientRects": false,
    "spoofFontFingerprint": false,
    "spoofMediaDevices": false,
    "timeZone": "ip",
    "trackingProtectionMode": "always",
    "webRTCPolicy": "disable_non_proxied_udp",
    "webSockets": "allow_all"
  },
  "profile": {
    "selected": "none",
    "interval": {
      "option": 0,
      "min": 1,
      "max": 1
    }
  },
  "version": "0.21.8.1",
  "whitelist": {
    "enabledContextMenu": false,
    "defaultProfile": "none",
    "rules": []
  },
  "settings": {
    "customScreen": "",
    "enableScriptInjection": true,
    "interval": 0,
    "kbDelay": "30",
    "limitHistory": false,
    "minInterval": null,
    "maxInterval": null,
    "notificationsEnabled": true,
    "protectKeyboardFingerprint": false,
    "protectWinName": false,
    "screenSize": "default",
    "spoofAudioContext": false,
    "spoofClientRects": false,
    "timeZone": "ip",
    "useragent": "real",
    "useragentValue": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:68.0) Gecko/20100101 Firefox/68.0",
    "webSockets": "allow_all"
  }
}
```

## Context (Environment)

Firefox 83.0, using Debian buster.

Note that everything started happening once I updated chameleon, everything was fine before. Twitch seems to be fixed when  spoof language to english is set.

**Comment by HanabishiRecca** [comment]
Can confirm this. Breakage happens regardless of the settings. Tons of errors in the console, seems like this is CSP-related issue.

**Comment by sereneblue** [comment]
@Minimalist73 @HanabishiRecca Please let me know if v0.21.9 resolves this issue.

**Comment by Minimalist73** [comment]
@sereneblue Hi,

Thanks for the quick fix, it seems to be good for both Twitch and google. I still get the blank page problem on reddit, it load the page and then just vanish. I tried with a clean Firefox with only uBlock & Chameleon with the same settings as the main message and I'm still getting the same result (debian buster too).

Here is what the console is telling me when I'm loading the Monero sub reddit (full logs):

```
18:14:21.194 downloadable font: rejected by sanitizer (font-family: "redesignFont" style:normal weight:400 stretch:100 src index:0) source: https://www.redditstatic.com/desktop2x/fonts/redesignIcon/redesignFont.5b316156767ef107284a5a23e2d700cc.eot
18:14:21.248 The resource at “https://platform.twitter.com/widgets.js” was blocked because content blocking is enabled. Monero
18:14:21.252 Uncaught ReferenceError: twttr is not defined
    onload https://www.redditmedia.com/mediaembed/kazt5r?responsive=true&is_nightmode=false:1
kazt5r:1:409
18:14:22.325
                  ,d"=≥,.,qOp,
                 ,7'  ''²$(  )
                ,7'      '?q$7'
             ..,$$,.
   ,.  .,,--***²""²***--,,.  .,
 ²   ,p²''              ''²q,   ²
:  ,7'                      '7,  :
 ' $      ,db,      ,db,      $ '
  '$      ²$$²      ²$$²      $'    Using Reddit at work? Work for Reddit.
  '$                          $'        https://www.reddit.com/jobs
   '$.     .,        ,.     .$'
    'b,     '²«»«»«»²'     ,d'
     '²?bn,,          ,,nd?²'
       ,7$ ''²²²²²²²²'' $7,
     ,² ²$              $² ²,
     $  :$              $:  $
     $   $              $   $
     'b  q:            :p  d'
      '²«?$.          .$?»²'
         'b            d'
       ,²²'?,.      .,?'²²,
      ²==--≥²²==--==²²≤--==²
index.ts:5:10
18:14:22.648 Starting Raven release e6b957c-production public url https://6b5ce051b2ed4032bd906af3de0630cd@oops.redditmedia.com/11 index.tsx:185:10
18:14:23.324 Loading failed for the <script> with source “https://c.aaxads.com/aax.js?pub=AAX763KC6&hst=www.reddit.com&ver=1.2”. index.m.less:1
18:14:23.638 Specified “type” attribute of “application/vnd.apple.mpegURL” is not supported. Load of media resource https://v.redd.it/yv0pjwtq28461/HLSPlaylist.m3u8?a=1610298859%2CNGI0NjcwZWIwNjJiZDUxZmQwZDY1NTQwOWFkOTU0NWFiZTI2YzQ1ZmI2OTk1ZjQ0NjVlOGRmN2QwZTYzOGEzOQ%3D%3D&v=1&f=sd failed. Monero
18:14:23.646 All candidate resources failed to load. Media load paused. Monero
18:14:23.876 Content Security Policy: Ignoring ‘x-frame-options’ because of ‘frame-ancestors’ directive.
18:14:23.923 XML Parsing Error: syntax error
Location: https://www.reddit.com/r/Monero/
Line Number 1, Column 1: index.m.less:1
18:14:24.101 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.323 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.325 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.329 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.330 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.331 TypeError: this[spoofContext.Skjnx6h] is undefined
    toDateString (index):295
    toString (index):301
    a index.ts:5
    ae index.tsx:383
    React 14
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.331 TypeError: this[spoofContext.Skjnx6h] is undefined
    getFullYear (index):188
    a index.tsx:124
    React 11
    executeBatch index.ts:199
    r helpers.js:74
react-dom.production.min.js:187:194
18:14:24.335 TypeError: this[spoofContext.Skjnx6h] is undefined one_tap:295:5
18:14:24.340 Specified “type” attribute of “application/vnd.apple.mpegURL” is not supported. Load of media resource https://v.redd.it/yv0pjwtq28461/HLSPlaylist.m3u8?a=1610298859%2CNGI0NjcwZWIwNjJiZDUxZmQwZDY1NTQwOWFkOTU0NWFiZTI2YzQ1ZmI2OTk1ZjQ0NjVlOGRmN2QwZTYzOGEzOQ%3D%3D&v=1&f=sd failed. Monero
18:14:24.379 All candidate resources failed to load. Media load paused. Monero
```

**Comment by sereneblue** [comment]
@Minimalist73 Thanks for the log. That looks to be a bug with timezone spoofing. Hopefully, it should be a simple fix.

**Comment by BarRunBoatHonestyOperation** [comment]
I went each setting one by one and looks like it's a timezone spoofing issue.  Here's what the Firefox debugger gives me when it breaks due to an exception.

` TypeError: this[spoofContext.Hnh4fmrxf] is undefined `

`  spoofContext.Date.prototype.toDateString = function() {
    if (isNaN(getTime.call(this))) {
      return "Invalid Date";
    }
    return toDateString.apply(this[spoofContext.Hnh4fmrxf].date);`

**Comment by sereneblue** [comment]
@BarRunBoatHonestyOperation @Minimalist73 This issue should be resolved with v0.21.10.

**Comment by Minimalist73** [comment]
@sereneblue It seems reddit is working fine now. Thanks for the update !
